### PR TITLE
Add default report data if empty product data is returned

### LINKED ISSUE #579: REST API: Products requests failing
When loading the products report, the REST API requests that are created to retrieve data for the summary numbers and chart ( 3 ) request, all fail with 500 errors from the API. This is only happening on time periods where no sale data exists.

**To Reproduce**

1. Open up your network tab on the js console
2. Visit the Products Report. Select a date range where **no** sales have been generated. I.E. a date range you haven't smooth-generated sales for.
3. Note all 3 fail
4. See error in php log:

```
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'totals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Warning:  get_object_vars() expects parameter 1 to be object, null given in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'intervals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Warning:  Invalid argument supplied for foreach() in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'total' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 70
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'totals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'totals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 71
[18-Oct-2018 19:15:48 UTC] PHP Warning:  get_object_vars() expects parameter 1 to be object, null given in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Warning:  get_object_vars() expects parameter 1 to be object, null given in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 60
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'page_no' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 73
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'intervals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 74
[18-Oct-2018 19:15:48 UTC] PHP Warning:  Invalid argument supplied for foreach() in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'total' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 70
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'intervals' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 71
[18-Oct-2018 19:15:48 UTC] PHP Warning:  Invalid argument supplied for foreach() in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 64
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'page_no' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 73
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'total' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 70
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 74
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 71
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'page_no' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 73
[18-Oct-2018 19:15:48 UTC] PHP Notice:  Trying to get property 'pages' of non-object in /srv/www/example/public_html/wp-content/plugins/wc-admin/includes/api/class-wc-admin-rest-reports-products-stats-controller.php on line 74
```

**Expected behavior**
Expect the API requests to return empty data sets when no data exists for the queried range.

**Details**
Happening in the `master` branch.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #579 

Fixes the API request and error displayed when a date is set where no order history is available.

### Screenshots

screen shot 2018-10-18 at 4 49 57 pm
screen shot 2018-10-18 at 4 50 05 pm

### Detailed test instructions:

1.  Navigate to `/wp-admin/admin.php?page=wc-admin#/analytics/products`
2.  Select dates where no order history is relevant and verify that empty chart is displayed and no network errors occur.
3.  Select dates where order history is available and verify that relevant chart data shows as expected.

### CODE REVIEW

**Review by Aljullu — COMMENTED** [review]
That fixes the issue! :clap:  I added a couple of minor comments in the code.

I see in the first screenshot that we will have to fix the chart, because it looks broken when there is no data. But that's for another PR. :slightly_smiling_face:

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php [hunk 1]
```diff
@@ -213,6 +220,28 @@ public function get_item_schema() {
...
 	}
 
+	/**	
+	 * Set the default results to 0 if API returns an empty array
+	 * 
+	 * @param $results mixed
+	 * @param $args array
+	 * @return object
+	 */
+	public function set_default_report_data( $results, $args ) {
```
Comment (Aljullu) [inline includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: `$args` is not needed here, right?

```suggestion
	public function set_default_report_data( $results ) {
```

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php [hunk 2]
```diff
@@ -213,6 +220,28 @@ public function get_item_schema() {
 		return $this->add_additional_fields_schema( $schema );
 	}
 
+	/**	
+	 * Set the default results to 0 if API returns an empty array
+	 * 
+	 * @param $results mixed
```
Comment (Aljullu) [inline includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: It looks like there is something wrong with this doc comments:
https://travis-ci.org/woocommerce/wc-admin/builds/443393089

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php (hunk 1 above)
Comment (joshuatf) [reply includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: You're right, we can remove this.

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php (hunk 2 above)
Comment (joshuatf) [reply includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: Whoops, thought I pushed this change already.  Should be okay now 😄

**Comment by joshuatf** [comment]
Thanks for taking the time to review, @Aljullu !  

Agreed on the empty chart issues; I did my best to fix up the header data, but I think the chart needs its own PR.  Opened an issue here - https://github.com/woocommerce/wc-admin/issues/582

Let me know if everything looks okay now.

**Review by Aljullu — APPROVED** [review]
Thanks for applying the changes, good work!

It looks good to me now. :+1:

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php [hunk 3]
```diff
@@ -31,6 +31,13 @@ class WC_Admin_REST_Reports_Products_Stats_Controller extends WC_REST_Reports_Co
 	 */
 	protected $rest_base = 'reports/products/stats';
 
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'woocommerce_reports_products_stats_select_query', array( $this, 'set_default_report_data' ), 10, 2 );
```
Comment (ryelle) [inline includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: If you're not using `$args` now, you can remove the `10, 2` here & fall back to the default value of 1 accepted arg (and default priority 10).

**Review by ryelle — CHANGES_REQUESTED** [review]
One small change, otherwise works well 👍

File: includes/api/class-wc-admin-rest-reports-products-stats-controller.php (hunk 3 above)
Comment (joshuatf) [reply includes/api/class-wc-admin-rest-reports-products-stats-controller.php]: Thanks, Kelly!  Removed.

**Comment by joshuatf** [comment]
@ryelle Updated with your change.  Can you take a look and let me know if everything looks okay now?  Thanks!

**Review by ryelle — APPROVED** [review]
Looks good now, thanks for updating 🙂

**Pull request merged**